cloud-controller-manager/do/firewall_controller_test.go

```diff
@@ -310,407 +115,178 @@ func TestFirewallController_Get(t *testing.T) {
-			expectedGodoFirewallListResp: func(context.Context, *godo.ListOptions) ([]godo.Firewall, *godo.Response, error) {
-				return nil, newFakeNotOKResponse(), errors.New("unexpected error")
-			},
-			expectedError: errors.New("failed to get worker firewall: failed to retrieve list of firewalls from DO API"),
-		},
-		{
-			name:    "when firewall matches what we expect there is nothing to reconcile",
-			fwCache: newFakeFirewallCache(),
-			expectedGodoFirewallGetResp: func(context.Context, string) (*godo.Firewall, *godo.Response, error) {
-				return newFakeFirewall(), newFakeOKResponse(), nil
-			},
-		},
-		{
-			name:    "when firewall needs to be reconciled an update call is made",
-			fwCache: newFakeFirewallCacheWithDiffInboundRules(),
-			expectedGodoFirewallGetResp: func(context.Context, string) (*godo.Firewall, *godo.Response, error) {
-				return newFakeFirewall(), newFakeOKResponse(), nil
-			},
-			expectedGodoFirewallUpdateResp: func(context.Context, string, *godo.FirewallRequest) (*godo.Firewall, *godo.Response, error) {
-				return newFakeFirewallWithDiffInboundRules(), newFakeOKResponse(), nil
-			},
-		},
-		{
-			name:    "error is returned when firewall cache exists but needs to be reconciled and update call fails",
-			fwCache: newFakeFirewallCacheWithDiffInboundRules(),
-			expectedGodoFirewallGetResp: func(context.Context, string) (*godo.Firewall, *godo.Response, error) {
-				return newFakeFirewall(), newFakeOKResponse(), nil
-			},
-			expectedGodoFirewallUpdateResp: func(context.Context, string, *godo.FirewallRequest) (*godo.Firewall, *godo.Response, error) {
-				return nil, newFakeNotOKResponse(), errors.New("unexpected error")
-			},
-			expectedError: errors.New("failed to reconcile worker firewall: failed to set reconciled firewall: could not update firewall: unexpected error"),
-		},
 	}
 
-	for _, test := range testcases {
-		t.Run(test.name, func(t *testing.T) {
-			// setup
-			gclient := newFakeGodoClient(
-				&fakeFirewallService{
-					listFunc:   test.expectedGodoFirewallListResp,
-					updateFunc: test.expectedGodoFirewallUpdateResp,
-					createFunc: test.expectedGodoFirewallCreateResp,
-					getFunc:    test.expectedGodoFirewallGetResp,
-				},
-			)
-			fwManager := newFakeFirewallManager(gclient, test.fwCache)
-			fc := NewFirewallController(ctx, kclient, gclient, inf.Core().V1().Services(), fwManager, testWorkerFWTags, testWorkerFWName, fwManager.metrics)
-
-			err := fc.reconcileCloudFirewallChanges(ctx)
-			if (err != nil && test.expectedError == nil) || (err == nil && test.expectedError != nil) {
-				t.Errorf("error with firewall controller run\nwant: %#v\n got: %#v", test.expectedError, err)
-			}
-		})
-	}
-}
+	eg.Go(loopCreator(func() error {
+		_, err := fc.ensureReconciledFirewall(ctx)
+		if err != nil {
+			return fmt.Errorf("ensureReconciledFirewall failed: %s", err)
+		}
+		return nil
+	}))
 
-func TestFirewallController_isEqualCheckForFirewallRequest(t *testing.T) {
```

Comment:
Note to reviewers: the method and test have been superseded by the new improved isEqual / diff functionality located in `firewall_controller_diff{_test}.go`.